gentoopm/portagepm/pkg.py

```diff
@@ -307,19 +328,39 @@ def repository(self):
         return self._dbapi.getRepositoryName(self._tree)
 
     @property
-    def maintainers(self):
-        # yes, seriously, the only API portage has is direct parser
-        # for the XML file
-        xml_path = os.path.join(os.path.dirname(self.path), "metadata.xml")
-        try:
-            meta = MetaDataXML(xml_path, None)
-        except (IOError, OSError) as e:
-            if e.errno == errno.ENOENT:
-                return ()
-            raise
-
+    def maintainers(self) -> typing.Optional[Iterable[PortagePackageMaintainer]]:
+        if (meta := self._metadata) is None:
+            return None
         return tuple(PortagePackageMaintainer(m) for m in meta.maintainers())
 
+    @property
+    def upstream(self) -> typing.Optional[PMUpstream]:
```

Comment:
To be honest, this isn't really consistent with gentoopm design. The idea was to provide thin wrappers over PM data types, not copy everything over. So in this case you'd return a `PortageUpstream` object that would provide accessors for all the shit.

Or well, technically, you wouldn't make a special "upstream" subclass about it and just put them as top-level properties of package class.

Reply:
Did I get it right this time?